model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProvider.java

```diff
@@ -165,40 +160,18 @@ private EventEntity convertEvent(Event event) {
         eventEntity.setIpAddress(event.getIpAddress());
         eventEntity.setError(event.getError());
         try {
-            if (maxDetailLength > 0 && event.getDetails() != null) {
-                Map<String, String> result = new HashMap<>(event.getDetails());
-                result.entrySet().forEach(t -> t.setValue(trimToMaxDetailLength(t.getValue())));
-
-                eventEntity.setDetailsJson(trimToMaxFieldLength(mapper.writeValueAsString(result)));
+            String details = mapper.writeValueAsString(event.getDetails());
+            if (details == null || details.length() <= 2550) { // 2550 is the lenght of the DETAILS_JSON field in EventEntity
```

Comment:
Can you define the constant as static final variable and use it on second occurrence as well?